python/ray/air/config.py

```diff
@@ -622,3 +629,6 @@ def __repr__(self):
                 "checkpoint_config": CheckpointConfig(),
             },
         )
+
+    def _repr_html_(self) -> str:
+        return make_table_html_repr(obj=self)
```

Comment:
The formatting looks kinda jank here. If it's not too difficult, could you fix it? 

<img width="1581" alt="image" src="https://user-images.githubusercontent.com/26107013/182730856-747cec7a-509a-4ea3-86e9-5e8b74fdf67e.png">

If it's complicated to fix, I think it's okay to leave as-is for now.

Reply:
Also, a suggestion: what do you think about making "Failure Config", "Sync Config", and "Checkpoint Config" subsections of the "Run Config" table? I feel like it might be easier to read if we don't nest the tables. It'd also fix the jank formatting problem.

Reply:
Ahh, thanks for pointing this out! The only reason I didn't do it before was that `BaseTrainer.run_config` used to accept a `dict` of objects, so I couldn't know _a priori_ what the keys and values were. But after rebasing onto `master`, I see now the type is `Optional[RunConfig]`, so I'll create a dedicated `_repr_html_` for this.

Reply:
Okay, I've reworked this to have the general settings for the `RunConfig` at the top, and then a few dedicated subsections for the sync, checkpoint, and failure configs. What do you think?

![image](https://user-images.githubusercontent.com/14017872/182999575-317cbcc0-8a78-43a2-aacc-c51a9ae5c4d5.png)

Reply:
This looks great!